### ISSUE TITLE
inconsistent error reporting by plugin and detekt

### ISSUE THREAD

**Issue by os97673** [detekt/detekt-intellij-plugin#504]
I have
```
potential-bugs:
  active: true
  severity: error
  ImplicitDefaultLocale:
    active: true
```
in my detekt.yml and when I run detekt it reports problem in the code as error
![image](https://github.com/detekt/detekt-intellij-plugin/assets/450628/de7a7e9a-416b-4a21-b740-9a56c6007bfc)
but in the editor (Android Studio Hedgehog | 2023.1.1 Patch 1) the problem is shown as a warning :-(
<img width="797" alt="Screenshot 2024-01-04 at 7 44 15 AM" src="https://github.com/detekt/detekt-intellij-plugin/assets/450628/36ec32f3-6a60-426f-b1f8-03d4a7c795de">

The detekt plugin (2.3.0) is configured to use the config, but it looks like it overrides severity with "treat detekt findings as errors" (which is not checked by my because I don't want ALL findings from the config to be reported as error)
Is this expected behavior or is this a bug?

**Comment by os97673** [comment]
I think the "problem" is in the way "treat as error" handled (https://github.com/detekt/detekt-intellij-plugin/blob/e808894ee69bc746446b353fb138a84caf6b9281/src/main/kotlin/io/gitlab/arturbosch/detekt/idea/DetektAnnotator.kt#L60C32-L60C54)
I think instead of
```
val severity = if (settings.treatAsErrors) HighlightSeverity.ERROR else HighlightSeverity.WARNING
```
it should be something like
```
val severity = if (settings.treatAsErrors) HighlightSeverity.ERROR else convertSeverity(finding.severity)
```
Or, at least, do this for the case when user specified a custom detekt config. Of course if it is ok to make such "backward incompatible" change.

Would you be open for such PR?

**Comment by schalkms** [comment]
Closed with [0df1e1a](https://github.com/detekt/detekt-intellij-plugin/commit/0df1e1a145b733e447f18b4dd2a184c2c91e6e87)
